pkg/operator/starter.go

```diff
@@ -91,15 +102,97 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		return err
 	}
 
+	caSyncController, err := newCustomCABundleSyncer(
+		operatorClient,
+		kubeInformersForNamespaces,
+		kubeClient,
+		controllerConfig.EventRecorder,
+	)
+	if err != nil {
+		return fmt.Errorf("could not create the custom CA bundle syncer: %w", err)
+	}
+
 	klog.Info("Starting the informers")
 	go kubeInformersForNamespaces.Start(ctx.Done())
 	go dynamicInformers.Start(ctx.Done())
 	go configInformers.Start(ctx.Done())
 
 	klog.Info("Starting controllerset")
 	go csiControllerSet.Run(ctx, 1)
+	go caSyncController.Run(ctx, 1)
 
 	<-ctx.Done()
 
 	return fmt.Errorf("stopped")
 }
+
+type controllerTemplateData struct {
+	CABundleConfigMap string
+}
+
+// withCustomCABundle executes the asset as a template to fill out the parts required when using a custom CA bundle.
+// The `caBundleConfigMap` parameter specifies the name of the ConfigMap containing the custom CA bundle. If the
+// argument supplied is empty, then no custom CA bundle will be used.
+func withCustomCABundle(assetFunc func(string) []byte, kubeClient kubeclient.Interface) func(string) []byte {
+	templateData := controllerTemplateData{}
+	switch used, err := isCustomCABundleUsed(kubeClient); {
+	case err != nil:
+		klog.Fatalf("could not determine if a custom CA bundle is in use: %v", err)
```

Comment:
This will panic on random API server errors, which is bad.
@bertinatto do you have idea how to work around this?

Reply:
Ideally we shouldn't do any replacement during the start up. That's better done in a controller